tss-esapi/tests/integration_tests/attributes_tests/session_attributes_tests.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2022 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use tss_esapi::{
+    attributes::{SessionAttributes, SessionAttributesBuilder, SessionAttributesMask},
+    tss2_esys::TPMA_SESSION,
+};
+
+macro_rules! test_valid_conversion {
+    ($tpm_value:expr, $method:ident) => {
+        let tpma_session_attribute: TPMA_SESSION = $tpm_value;
+        let session_attributes = SessionAttributes::from(tpma_session_attribute);
+        assert_eq!(
+            true,
+            session_attributes.$method(),
+            "SessionAttributes converted from TPMA_SESSION = {} did not produce the expected result with regard to {}.", tpma_session_attribute, stringify!($method),
+        );
```

Comment:
This can probably be changed to 
```
assert!(
            session_attributes.$method(),
            "SessionAttributes converted from TPMA_SESSION = {} did not produce the expected result with regard to {}.", tpma_session_attribute, stringify!($method),
        );
```

Reply:
Isn't this exactly what the `assert!` macro is doing? I'm reading https://doc.rust-lang.org/std/macro.assert.html and it says `Asserts that a boolean expression is true at runtime.`.

Reply:
Yeah, but I am not testing an expression. I am testing a return value that in this case happen to be a bool.

so with assert, you will get a message saying like assertion error looking something like this:
```thread 'main' panicked at 'assertion failed: some_computation()', src/main.rs:9:1```

But with assert_eq you get:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`', src/main.rs:9:1
 ```
 
 To me the latter shows more clearly what I expected and what I got both in the code and in the message.

Reply:
I think there are good arguments for both. I like seeing what was expected and what we got, but it would be nice if it actually specified which one is which so we don't have to look at the code, but such is life. Leave it as is!

Reply:
Yes, it would have been nice if it would have been 'expected' and 'actual' as it is in google test for c++ for example. I wonder if there is a crate that cam creates asserts for testing that are more like that?